src/esp/assets/PTexMeshData.cpp

```diff
@@ -798,6 +820,8 @@ void PTexMeshData::uploadBuffersToGPU(bool forceReload) {
                                       Magnum::GL::BufferUsage::StaticDraw);
     currentMesh->indexBuffer.setData(submeshes_[iMesh].ibo,
                                      Magnum::GL::BufferUsage::StaticDraw);
+    currentMesh->triangleMeshIndexBuffer.setData(
```

Comment:
Is it true that we're using additional CPU and GPU memory for this triangle index buffer, for all PTexMesh's, regardless of whether the picking feature is being used? Does this increase our memory usage during RL training?

Reply:
Plus it is just the index here. Yes, it will increase the memory usage, but compared to the HDR textures used in the replica mesh, it is not significant for sure. 
So I would leave it like this to keep the code simple.

Reply:
Can you create a Task and link it here in the code in a comment? Or some other way to remember this as a candidate for optimization.

From Brennan's recent presentation on batched rendering, it sounded like GPU memory is the bottleneck limiting further batching, so I'd at least like to remember this is here to be optimized later.

Reply:
@eundersander :

I do not think it is necessary.
For the entire replica dataset, I printed out the memory usage of this index buffer "ibo_tri" for each model, list it side-by-side with the texture size in the following table:

NAME   ====     SIZE         ====        TEXTURE SIZE                                                         
apartment_0 ==== 104MB (0.10GB)  ====      7GB                                                                
apartment_1====  43MB (0.04GB)   ====       2.9GB                                                              
apartment_2 ==== 48MB (0.05GB)   ====       3.3GB                                                              
frl_apartment_0 ==== 40MB (0.04GB) ====     2.7GB                                                              
frl_apartment_1 ==== 40MB (0.04GB) ====     2.7GB                                                              
frl_apartment_2 ==== 41MB (0.04GB) ====     2.8GB                                                              
frl_apartment_3 ==== 40MB (0.04GB) ====     2.7GB                                                              
frl_apartment_4 ==== 41MB (0.04GB)   ====   2.8GB                                                              
frl_apartment_5====  40MB (0.04GB) ====     2.7GB                                                              
hotel_0    ====      25MB (0.02GB) ====     1.7GB                                                              
office_0   ====      13MB (0.01GB) ====     906.9MB                                                            
office_1   ====      9MB (0.009GB)  ====    650MB                                                              
office_2    ====     19MB (0.02GB)  ====    1.3GB                                                              
office_3      ====   27MB (0.03GB) ====     1.8GB                                                              
office_4    ====     22MB (0.02GB)   ====   1.5GB                                                              
room_0     ====      21MB (0.02GB)   ====   1.5GB                                                              
room_1     ====      14MB (0.01GB)  ====    993.8MB                                                            
room_2      ====     16MB (0.02GB) ====      1.1GB 

The size of the texture is roughly **70x** of the new index buffer size.
"GPU memory is the bottleneck limiting further batching" -- it makes sense. 
However, the index buffer is certainly not the "bottleneck" here.

Reply:
This is great data. I'm not asking you to make any code changes. I'm just asking for a code comment or something, to help the future engineer working on memory optimization. "beware this ibo_tri buffer consumes ~20 MB GPU memory per mesh and is only used for object-picking". If this is in the ballpark of 1% of total GPU mem usage, it might be worth pursuing in the future.

Reply:
Sure, I can add comments here. This is good suggestion.
In terms of optimization, this would be the very last place I will look into since the space complexity for the index buffer and the texture are not at the same level at all.